internal/api/v1/services.go

```diff
@@ -136,6 +140,284 @@ func (sc ServicesController) Index(w http.ResponseWriter, r *http.Request) {
 	}
 }
 
+func (sc ServicesController) CreateCustom(w http.ResponseWriter, r *http.Request) {
+	params := httprouter.ParamsFromContext(r.Context())
+	org := params.ByName("org")
+
+	defer r.Body.Close()
+	bodyBytes, err := ioutil.ReadAll(r.Body)
+	if handleError(w, err, http.StatusInternalServerError) {
+		return
+	}
+
+	var createRequest models.CustomCreateRequest
+	err = json.Unmarshal(bodyBytes, &createRequest)
+	if handleError(w, err, http.StatusBadRequest) {
+		return
+	}
+
+	if createRequest.Name == "" {
+		http.Error(w, fmt.Sprintf("Cannot create custom service without a name"),
+			http.StatusBadRequest)
+		return
+	}
+
+	if len(createRequest.Data) < 1 {
+		http.Error(w, fmt.Sprintf("Cannot create custom service without data"),
+			http.StatusBadRequest)
+		return
+	}
+
+	gitea, err := clients.GetGiteaClient()
+	if handleError(w, err, http.StatusInternalServerError) {
+		return
+	}
+
+	exists, err := gitea.OrgExists(org)
+	if handleError(w, err, http.StatusInternalServerError) {
+		return
+	}
+	if !exists {
+		http.Error(w, fmt.Sprintf("Organization '%s' does not exist", org),
+			http.StatusNotFound)
+		return
+	}
+
+	cluster, err := kubernetes.GetCluster()
+	if handleError(w, err, http.StatusInternalServerError) {
+		return
+	}
+
+	// Verify that the requested name is not yet used by a different service.
+	_, err = services.Lookup(cluster, org, createRequest.Name)
+	if err == nil {
```

Comment:
the opposite is not always safe to ignore. I mean, if there is an error, it doesn't mean the error is related to the service not being there, it may be some irrelevant error which shouldn't be ignored.

You need to check what kind of error this was.

Reply:
Checking...

Reply:
Done. Commit 3477a08cded8b3d70df5e707d00bb0d461afa91f